sdk/keyvault/Azure.Security.KeyVault.Certificates/src/LightweightPkcs8Decoder.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+
+namespace Azure.Core
+{
+    internal partial class LightweightPkcs8Decoder
```

Comment:
Why not put this method into the main one?

Reply:
For now I don't want to risk any ECDsa support in Azure.Identity. To create an ECDsa certificate requires calling APIs declared by netstandard2.0, but not actually defined by net461. In Key Vault I had to work around this in various ways and don't want to put Azure.Identity at risk, especially since ECDsa certificates won't satisfy `ClientCertificateCredential`: it requires an RSA certificate.

Reply:
Originally I even defined the methods that weren't already defined in the shared `LightweightPkcs8Decoder` in Key Vault's copy, but they are general purpose enough and I added enough tests to feel confident in our very limited support for necessary ASN.1 tags. I figured this way no one else has to test them should they be needed (like decoding OIDs).

As defined here, this was all about reducing risk exposure within reason.